pkg/kstatus/polling/aggregator/aggregator.go

```diff
@@ -35,43 +47,41 @@ func NewAllCurrentOrNotFoundStatusAggregator(identifiers []wait.ResourceIdentifi
 // ResourceStatus is called whenever we have an observation of a resource. In this
 // case, we just keep the latest status so we can later compute the aggregate status
 // for all the resources.
-func (d *BasicAggregator) ResourceStatus(r *event.ResourceStatus) {
-	d.resourceCurrentStatus[r.Identifier] = r.Status
+func (g *genericAggregator) ResourceStatus(r *event.ResourceStatus) {
+	g.resourceCurrentStatus[r.Identifier] = r.Status
 }
 
-// AggregateStatus computes the aggregate status for all the resources. In this
-// implementation of the Aggregator, we treat resources with the NotFound status as Current.
-func (d *BasicAggregator) AggregateStatus() status.Status {
-	// if we are not observing any resources, we consider status be Current.
-	if len(d.resourceCurrentStatus) == 0 {
-		return status.CurrentStatus
+// AggregateStatus computes the aggregate status for all the resources.
+func (g *genericAggregator) AggregateStatus() status.Status {
+	if len(g.resourceCurrentStatus) == 0 {
+		return g.desiredStatus
 	}
 
-	allCurrentOrNotFound := true
+	allDesired := true
 	anyUnknown := false
-	for _, s := range d.resourceCurrentStatus {
+	for _, s := range g.resourceCurrentStatus {
 		if s == status.FailedStatus {
 			return status.FailedStatus
 		}
 		if s == status.UnknownStatus {
```

Comment:
could just return status.UnknownStatus here, right?

Reply:
So the comment for this method is:
```
// AggregateStatus returns an aggregation of resource status per the following rules.
// - If any of the resources ...
```
just copy paste your fine comment.

Reply:
Done